pkgs/tools/X11/xdg-utils/default.nix

```diff
@@ -15,73 +16,233 @@ let
     sha256 = "15gw2nyrqmdsdin8gzxihpn77grhk9l97jp7s7pr7sl4n9ya2rpj";
   };
 
-  perlPath = with perlPackages; makePerlPath [
-    NetDBus XMLTwig XMLParser X11Protocol
+  # Required by the common desktop detection code
+  commonDeps = [ dbus coreutils gnugrep gnused ];
+  commonFakes = [
+    "explorer.exe"
+    "gnome-default-applications-properties"
+    "kde-config"
+    "xprop"
   ];
 
+  # This is still required to work around the eval trickery some scripts do
+  commonPrologue = "${writeText "xdg-utils-prologue" ''
+    export PATH=$PATH:${coreutils}/bin
+  ''}";
+
+  solutions = [
+    {
+      scripts = [ "bin/xdg-desktop-icon" ];
+      interpreter = "${bash}/bin/bash";
+      inputs = commonDeps;
+      fake.external = commonFakes ++ [
+        "gconftool-2"
+        "xdg-user-dir"
+      ];
+      keep."$KDE_SESSION_VERSION" = true;
+      prologue = commonPrologue;
+    }
+
+    {
+      scripts = [ "bin/xdg-desktop-menu" ];
+      interpreter = "${bash}/bin/bash";
+      inputs = commonDeps ++ [ gawk ];
+      fake.external = commonFakes;
+      keep."$KDE_SESSION_VERSION" = true;
+      prologue = commonPrologue;
+    }
+
+    {
+      scripts = [ "bin/xdg-email" ];
+      interpreter = "${bash}/bin/bash";
+      inputs = commonDeps ++ [ gawk "${placeholder "out"}/bin" ];
+      execer = [
+        "cannot:${placeholder "out"}/bin/xdg-mime"
+        "cannot:${placeholder "out"}/bin/xdg-open"
+      ];
+      fake.external = commonFakes ++ [
+        "exo-open"
+        "gconftool-2"
+        "gdbus"
+        "gio"
+        "gnome-open"
+        "gvfs-open"
+        "qtxdg-mat"
+        "xdg-email-hook.sh"
+      ];
+      fix."/bin/echo" = true;
+      keep = {
+        "$command" = true;
+        "$kreadconfig" = true;
+        "$THUNDERBIRD" = true;
+        "$utf8" = true;
+      };
+    }
+
+    {
+      scripts = [ "bin/xdg-icon-resource" ];
+      interpreter = "${bash}/bin/bash";
+      inputs = commonDeps;
+      fake.external = commonFakes;
+      keep."$KDE_SESSION_VERSION" = true;
+      prologue = commonPrologue;
+    }
+
+    {
+      scripts = [ "bin/xdg-mime" ];
+      interpreter = "${bash}/bin/bash";
+      inputs = commonDeps ++ [ file gawk ];
+      fake.external = commonFakes ++ [
+        "gio"
+        "gnomevfs-info"
+        "gvfs-info"
+        "kde4-config"
+        "kfile"
+        "kmimetypefinder"
+        "kmimetypefinder5"
+        "ktraderclient"
+        "ktradertest"
+        "mimetype"
+        "qtpaths"
+        "qtxdg-mat"
+      ];
+      fix."/usr/bin/file" = true;
+      keep = {
+        "$KDE_SESSION_VERSION" = true;
+        "$KTRADER" = true;
+      };
+    }
```

Comment:
xdg-mime doesn't have a prologue, but it does eval both `kbuildsycoca*` and `cp`. 

(It also technically evals `$x/update-mime-database`, but that's in a loop so I'll call it out in ~degenerately-dynamic territory. Perhaps worth documenting, but probably not worth holding this up.)

I see references elsewhere to `${pkgs.libsForQt5.kservice}/bin/kbuildsycoca5` and `${pkgs.plasma5Packages.kservice}/bin/kbuildsycoca5`

Reply:
Missed that, added the prologue, kbuildsycoca is KDE stuff that we should not be pulling in.